packages/flame_tiled/lib/src/tiled_component.dart

```diff
@@ -12,20 +12,42 @@ import 'package:tiled/tiled.dart';
 /// It uses a preloaded [RenderableTiledMap] to batch rendering calls into
 /// Sprite Batches.
 /// {@endtemplate}
-class TiledComponent<T extends FlameGame> extends Component with HasGameRef<T> {
+class TiledComponent<T extends FlameGame> extends PositionComponent
+    with HasGameRef<T> {
   /// Map instance of this component.
   RenderableTiledMap tileMap;
 
-  /// The logical size of the component. The game assumes that this is the
-  /// approximate size of the object that will be drawn on the screen.
-  late final Vector2 size = _computeSize();
+  /// This property **cannot** be reassigned at runtime. To make the
+  /// [PositionComponent] larger or smaller, change its [scale].
+  @override
+  set size(Vector2 size) {
+    // Intenitonally left empty
+  }
+
+  /// This property **cannot** be reassigned at runtime. To make the
+  /// [PositionComponent] larger or smaller, change its [scale].
+  @override
+  set width(double w) {
+    // Intenitonally left empty
+  }
+
+  /// This property **cannot** be reassigned at runtime. To make the
+  /// [PositionComponent] larger or smaller, change its [scale].
+  @override
+  set height(double h) {
+    // Intenitonally left empty
+  }
```

Comment:
You can mark these setters with `@internal` so that users are warned when they try to use them.

Reply:
Tiled Component is the public API for this package.  `@internal` would throw an error for exposing it in the public API, no?  If you add `TiledComponent` to a list of `PositionComponent`, you wouldn't get that warning.

from pub.dev api:
> the declaration is declared in a package's public API, or is exposed from a package's public API, or

Reply:
Oh, my bad. I forgot that this is flame_tiled 😅